posts/2022/09-10-graphite-stacked-prs.md

```diff
@@ -0,0 +1,186 @@
+
+[Graphite](https://graphite.dev) makes the process of working with stacked PRs much easier. It can rebase and update a whole stack in one command, is much better at resolving rebases than the standard tooling, and works wonderfully in the terminal.
+
+[You can read the Graphite documentation here](https://docs.graphite.dev)
+
+It posts the relationships between the PRs in the stack to GitHub as a comment so that reviewers can traverse the relationships.
+
+![A Graphite comment on a pull request](graphite_comment.png "A Graphite comment on a pull request")
+
+And it can show you the relationships in the terminal and lets us navigate up and down the stack with commands like `gt up` and `gt down 3`
+
+And the free version is more than enough to reap the benefits.
+
+## The benefits
+
+### Less context switching and more flow
+
+Being able to stay on the same task is huge for me. A big part of developer happiness is the experience of being able to work for long stretches on challenging problems. This workflow enables us to batch up coding sessions and deployment sessions and does not leave you blocked by reviews. As a result, I felt motivated to write more code as I didn't feel like a brick wall was waiting for me.
+
+### More reviews
+
+Pull request reviews generally come in bursts. There will be long periods of no movement punctuated by sudden acts of mass review. This behaviour matches up with the common preference of batching up work. Therefore, the more PRs we can have ready when the next reviewer comes along, the more throughput we can manage.
+
+### Safer deployments
+
+The order in which we deploy matters, especially when it comes to microservices.
+
+- We run the migration before deploying the related logic.
+- We deploy the new API server before the client.
+- We remove the calling code before deleting the deprecated function.
+- Small PRs mean small merges and small deployments. Each deployment can be a small safe step. The alternative could be a large merge and a high-risk deployment that needs to be micro-managed and monitored.
+
+### Smaller testing loops
+
+Each of those PRs will have to pass the continuous integration tests, so we write tests and verify functionality in tighter loops. We can also work on the next PR in the stack while the previous one runs through CI.
+
+### Better planning
+
+I've found that my tickets are getting more detailed, and I am not just describing my work in smaller chunks but also deployment instructions around VMs, migrations, and monitoring. It's easy to miss these things when tickets are for larger scopes of work.
+
+## Considerations and drawbacks
+
+### It helps when we are better organized
+
+Because you might be much farther ahead in the stack when a PR at the bottom of the pile gets merged, it is essential to note down actions that we need to take after merging like:
```

Comment:
```suggestion
Since you might be much farther ahead in the stack when a PR at the bottom of the pile gets merged, it is essential to note down actions that we need to take after merging like:
```